src/amber/controller/base.cr

```diff
@@ -32,5 +32,13 @@ module Amber::Controller
     def session
       context.session
     end
+
+    def halt!(status_code : Int32 = 200, content = "")
+      response.headers["Content-Type"] = "text/plain"
+      response.status_code = status_code
+      response.print content
+      response.close
+      context.halt = true
```

Comment:
I think it would make more sense to set `context.content = content` inside of halt and still have the pipeline handler print it.

Then instead of checking `context.halt` we could just check `context.content`. This is cleaner and won't try to print to response twice.